### DISCUSSION TITLE
API rule for two multi relations sharing a common element

### DISCUSSION THREAD

**Discussion by tomaskebrle** [pocketbase/pocketbase#1817] [Q&A] [answered]
So I have a collection `users` and `groups`, each user has a `groups` relation with all the groups he's part of. I want for users from the same collection to be able to read each other's data. In other words if the `@request.groups.id` shares a common element with `groups.id` (groups on the record of the user I am trying to read), i want the rule to pass.

**Comment by ganigeorgiev** [comment, marked as answer]
This seems similar to the request in https://github.com/pocketbase/pocketbase/discussions/1808.

You can try the `?=` ("at least one of equal") operator, aka:

```js
@request.auth.groups:length > 0 && @request.auth.groups.id ?= groups.id
```

**Reply by tomaskebrle** [reply]
Thank you, I have tried this and it didn't work turns out I made a mistake in the query and put just `=`
```typescript
const members = await pb.collection('users').getFullList<UsersResponse>(200, {
filter: `groups.id ?= "${id}"`
});
```
Leaving this here for someone who maybe made the same mistake

**Reply by ganigeorgiev** [reply]
_Side note to clear eventual misunderstanding: the above js code snippet suggest that you are allowing every user to search for any other users since the above is a **client-side filter**. If you want to allow **only users within the same group** to be able search each other, then the recommended way is to define the suggested `@request.auth.*` filter as "List/Search API" rule to your `users` collection (you can still use the client-side filter if you want to further narrow down the filtered users)._
